converters/google/convert.go

```diff
@@ -212,17 +278,26 @@ func (c *Converter) addCreateOrUpdate(rc tfplan.ResourceChange) error {
 
 		key := converted.Type + converted.Name
 
-		if existing, exists := c.assets[key]; exists {
-			// The existance of a merge function signals that this tf resource maps to a
-			// patching operation on an API resource.
-			if mapper.mergeCreateUpdate != nil {
-				converted = mapper.mergeCreateUpdate(existing.converterAsset, converted)
-			} else {
+		existing, exists := c.assets[key]
+		var existingConverterAsset *converter.Asset
+		if !exists && mapper.fetch != nil{
+			asset, err := mapper.fetch(&rd, c.cfg)
+			existingConverterAsset = &asset
+			if err != nil {
+				return errors.Wrap(err, "fetching asset")
+			}
+		} else if exists {
+			existingConverterAsset = &existing.converterAsset
+		}
```

Comment:
Is there any reason we can't structure this like the following? Same logic (I think) but doesn't need to extra `!exists` check and the variables are in scope less long.

```suggestion
		var existingConverterAsset *converter.Asset
		if existing, exists := c.assets[key]; exists {
			existingConverterAsset = &existing.converterAsset
		} else if mapper.fetch != nil{
			asset, err := mapper.fetch(&rd, c.cfg)
			if err != nil {
				return errors.Wrap(err, "fetching asset")
			}

			existingConverterAsset = &asset
		}
```

Reply:
I don't remember how I ended up with that particular flow but I think your version is cleaner & does the same thing (plus moving setting of `existingConverterAsset = &asset` until after the error check is better.)